src/content/guides/using-persistent-volumes-on-aws-with-efs/index.md

```diff
@@ -0,0 +1,126 @@
+weight = 50
+tags = ["tutorial"]
++++
+
+# Using Persistent Volumes on AWS with EFS
+
+If your cluster is running in the cloud on Amazon Web Services (AWS) the most common way to store data is using EBS volumes with the [dynamic provisioner](https://docs.giantswarm.io/guides/using-persistent-volumes-on-aws/). Sometimes EBS is not the optimal solution.
+
+The advantages of using EFS over EBS are:
+- EFS data can be accessed from all Availability Zone in the same region while EBS is tied to a single Availability Zone.
+- EFS has the capability to mount the same Persistent Volume to multiple pods at the same time using the ReadWriteMany [access mode](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes).
+- EFS will not hit the [AWS Instance Volume Limit](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/volume_limits.html) as it is a software mount and will avoid the [Impaired EBS](https://docs.giantswarm.io/guides/aws-impaired-volumes/) issue.
+- EFS mount times are better than EBS.
+
+If you need to use EFS to provision volumes, be advised:
+- All Kubernetes Persistent Volumes will be stored in the same EFS instance. You can deploy multiple provisioners per cluster, each having its own storage-class and EFS instance.
+- [EFS throughtput](https://docs.aws.amazon.com/efs/latest/ug/performance.html) need to be set up accordingly in order to not have performance issues. We only recommend Provisioned Throughput, and if you need high performance you will need EBS.
+- EFS backups are done with [AWS Backup](https://aws.amazon.com/backup/) and it does not have the snapshot feature of EBS.
+- You cannot limit the amount of data stored in an EFS volume. The requested value in Kubernetes is ignored.
+- EFS mount targets are limited to 1 subnet per Availability Zone. Each NodePool will create a different subnet per AZ, plan accordingly. 
+
+
+## Provision an EFS instance on AWS
+
+- Open the [AWS management](https://aws.amazon.com/console/) console in the account your cluster is located.
+- Select EFS from the services list.
+- Create a new EFS mount and select the VPC where your cluster is located.
+- Select the Availability Zone with the subnets where your instances are located and the security-groups of the workers.
+- Choose the throughput and performance mode, no file system policy or access points are needed.
+- Create the instance and note down the EFS instance id.
+
+## EFS Provisioner configuration file
+Create a file on your local machine named "efs-provisioner.yaml" with the following content:
+```yaml
+global:
+  deployEnv: production
+
+efsProvisioner:
+  efsFileSystemId: fs-90f935c8
+  awsRegion: eu-central-1
+  path: /
+  provisionerName: giantswarm.io/aws-efs
+  storageClass:
+    name: efs
+    isDefault: false
+    gidAllocate:
+      enabled: true
+      gidMin: 40000
+      gidMax: 50000
+    reclaimPolicy: Delete
+    mountOptions: []
+
+rbac:
+  create: true
+
+podSecurityPolicy:
+  enabled: true
+```
```

Comment:
Is it clear here which values have to be adjusted by the user to match their case? When in doubt, add a line underneath, like "Adapt the awsRegion and ... to fit your needs"

Reply:
It's a couple lines down there, is this enough?

> You will need to populate the **efsFileSystemId** and **awsRegion** parameters to match the configured values of the EFS instance.

Reply:
OK, thanks!